packages/components/src/Link/Link.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { ReactNode, forwardRef } from "react";
+import Clickable, { ClickableProps } from "../Clickable";
+
+type LinkHTMLElementProps = Omit<
+  React.AnchorHTMLAttributes<HTMLAnchorElement>,
+  "disabled"
+>;
+
+export type LinkProps = LinkHTMLElementProps & {
+  as?: "a" | React.ComponentType<any>;
+  /**
+   * The link contents or label.
+   */
+  children: ReactNode;
+  color?: ClickableProps<"a">["color"];
+  "data-testid"?: string;
+  size?: ClickableProps<"a">["size"];
+  variant?: ClickableProps<"a">["variant"];
+};
+
+/**
+ * Component for making links that navigate to a different page that is outside of the
+ * product (ie help desk or external resource).
```

Comment:
Couldn't a link also be to a location within the product?

Reply:
You probably know better than I do since I haven't worked in LP much. My interpretation is that the react router should be used for in-product navigation. I wouldn't want people to think that hey we should be using the new design system link component and miss the react router pattern when we're not actually trying to change that by adding the new component.

Reply:
> My interpretation is that the react router should be used for in-product navigation

I think this is generally correct currently.

My thinking is this is an aspect of LP right now, but not something that the design system should really be worried about. The design system shouldn't really have LP-isms baked into it (within reason).

But at the end of the day, we're talking about semantics in a comment 😄, so either way is fine.

Maybe we could handle both sides by making the comment something like

> Component for making anchor tags.
>
> Note: if using a routing component like react-router, you'll probably need to pass in its Link component
> via the `as` prop.

Reply:
I like that, thanks! 👍